### PR TITLE
Pz/types 010

### PR DESCRIPTION
**Pull request by p-zielinski**
#### New schemas:
- LoyaltiesListMemberRewardsRequestQuery
- LoyaltiesGetPointsExpirationRequestQuery
- LoyaltiesGetPointsExpirationResponseBody
- LoyaltiesListCardTransactionsRequestQuery
- LoyaltiesListCardTransactionsResponseBody
- LoyaltyCardTransactionsType
- SimpleLoyaltyVoucher
- LoyaltyCardTransaction
- LoyaltyCardTransactionsFields
- LoyaltiesExportCardTransactionsRequestBody
- LoyaltiesExportCardTransactionsResponseBody
- RewardAssignment
- Reward
- RewardTypeCoin
- RewardTypeMaterial
- RewardTypeCampaign
- RewardType
- LoyaltiesAddOrRemoveCardBalanceResponseBody
- LoyaltiesTransferPointsResponseBody
- LoyaltiesTransferPoints
- LoyaltiesTransferPointsRequestBody
- LoyaltiesListMemberRewardsResponseBody
- LoyaltiesAddOrRemoveCardBalanceRequestBody
- PointsExpirationTypes

#### Schemas changes
- /v1/loyalties/{campaignId}/members/{memberId}/balance
    - new request schema `LoyaltiesAddOrRemoveCardBalanceRequestBody` (old `8_req_add_remove_points_balance`)
    - new response schema `LoyaltiesAddOrRemoveCardBalanceResponseBody` (old `8_res_add_remove_points_balance`)
- /v1/loyalties/members/{memberId}/balance
    - new request schema `LoyaltiesAddOrRemoveCardBalanceRequestBody` (old `8_req_add_remove_points_balance`)
    - new response schema `LoyaltiesAddOrRemoveCardBalanceResponseBody` (old `8_res_add_remove_points_balance`)
- /v1/loyalties/{campaignId}/members/{memberId}/transfers
    - new request schema `LoyaltiesTransferPointsRequestBody` (old `8_req_transfer_loyalty_points`)
    - new response schema `LoyaltiesTransferPointsResponseBody` (old `8_obj_loyalty_card_object_non_expanded_categories`)
- /v1/loyalties/{campaignId}/members/{memberId}/transactions
    - new request schema `LoyaltiesListCardTransactionsRequestBody` (old `8_res_get_loyalty_card_transactions`)
- /v1/loyalties/members/{memberId}/transactions
    - new request schema `LoyaltiesListCardTransactionsResponseBody` (old `8_res_get_loyalty_card_transactions`)
- /v1/loyalties/members/{memberId}/transactions/export
    - new request schema `LoyaltiesExportCardTransactionsRequestBody` (old `8_req_create_loyalty_card_transactions_export`)
    - new response schema `LoyaltiesExportCardTransactionsResponseBody` (old `8_obj_export_transactions_object`)
- /v1/loyalties/{campaignId}/members/{memberId}/transactions/export
    - new request schema `LoyaltiesExportCardTransactionsRequestBody` (old `8_req_create_loyalty_card_transactions_export`)
    - new response schema `LoyaltiesExportCardTransactionsResponseBody` (old `8_obj_export_transactions_object`)
- /v1/loyalties/{campaignId}/members/{memberId}/points-expiration
    - new response schema `LoyaltiesGetPointsExpirationResponseBody` (old `8_res_get_points_expiration`)

### CODE REVIEW

**Comment by p-zielinski** [comment]
https://docs.voucherify.io/v2018-08-01-piotr-465/reference/loyalty-campaign-object

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+- PointsExpirationTypes
+
+#### Schemas changes
+- /v1/loyalties/{campaignId}/members/{memberId}/balance
+    - new request schema `LoyaltiesAddOrRemoveCardBalanceRequestBody` (old `8_req_add_remove_points_balance`)
+    - new response schema `LoyaltiesAddOrRemoveCardBalanceResponseBody` (old `8_res_add_remove_points_balance`)
+- /v1/loyalties/members/{memberId}/balance
+    - new request schema `LoyaltiesAddOrRemoveCardBalanceRequestBody` (old `8_req_add_remove_points_balance`)
+    - new response schema `LoyaltiesAddOrRemoveCardBalanceResponseBody` (old `8_res_add_remove_points_balance`)
+- /v1/loyalties/{campaignId}/members/{memberId}/transfers
```
Comment (marcin-slezak) [inline Changelog.md:38]: Missing body params attributes descriptions

image

Comment (marcin-slezak) [reply Changelog.md:38]: and response body attributes description: 
image

File: Changelog.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+    - new request schema `LoyaltiesTransferPointsRequestBody` (old `8_req_transfer_loyalty_points`)
+    - new response schema `LoyaltiesTransferPointsResponseBody` (old `8_obj_loyalty_card_object_non_expanded_categories`)
+- /v1/loyalties/{campaignId}/members/{memberId}/transactions
+    - new request schema `LoyaltiesListCardTransactionsRequestBody` (old `8_res_get_loyalty_card_transactions`)
+- /v1/loyalties/members/{memberId}/transactions
+    - new request schema `LoyaltiesListCardTransactionsResponseBody` (old `8_res_get_loyalty_card_transactions`)
+- /v1/loyalties/members/{memberId}/transactions/export
+    - new request schema `LoyaltiesExportCardTransactionsRequestBody` (old `8_req_create_loyalty_card_transactions_export`)
+    - new response schema `LoyaltiesExportCardTransactionsResponseBody` (old `8_obj_export_transactions_object`)
```
Comment (marcin-slezak) [inline Changelog.md:47]: I see some differences in `conditions` and `result` attributes... please, confirm if that is correct.

image

**Review by marcin-slezak — CHANGES_REQUESTED** [review]

File: Changelog.md (hunk 2 above)
Comment (p-zielinski) [reply Changelog.md:47]: that is correct, this filter is based on memberId from url param. we cannot add additional filters

**Review by marcin-slezak — APPROVED** [review]

**Pull request merged**